state-sync/diem-data-client/src/diemnet/logging.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Error;
+use diem_logger::Schema;
+use serde::Serialize;
+
+#[derive(Schema)]
+pub struct LogSchema<'a> {
+    name: LogEntry,
+    error: Option<&'a Error>,
+    event: Option<LogEvent>,
+    message: Option<&'a str>,
```

Comment:
I don't think you'll need a `message` field, since structured logging already provides this.

You can log e.g.

```
info!(Schema::new(...), "Here is my message with {} field", field);
```

This will handle formatting, and it'll be named as `message`.

Reply:
I think this is a matter of preference, but good to know that it'll automatically parse the string.